src/stores/ui.ts

```diff
@@ -1,33 +1,148 @@
 import { defineStore } from 'pinia'
 
-export type SnackbarState = 'not_shown' | 'visible' | 'dismissed'
+/**
+ * By default, the filters are shown on desktop, and hidden on the mobile widths.
+ * `dismissed` is used when the filters are closed on the desktop width.
+ */
+export const VISIBLE = 'visible'
+export const DISMISSED = 'dismissed'
+export const NOT_SHOWN = 'not_shown'
+const snackbarStates = [NOT_SHOWN, VISIBLE, DISMISSED] as const
+export type SnackbarState = typeof snackbarStates[number]
```

Comment:
I have been thinking about this pattern a bit and I think we are probably slightly over-engineering this pattern (and not relying on our nice TypeScript tools) by defining constants and then using them.

If we could rely just on TypeScript, we could do just this:

```suggestion
const snackbarStates = ['not_shown', 'visible', 'dismissed'] as const
export type SnackbarState = typeof snackbarStates[number]
```

And then anywhere that accepts `SnackbarState`, auto-complete hints will show you the strings that are valid. What's more, in refactoring, if the SnackbarState changes to no longer accept some value, TypeScript will tell us this in type checking.

Following this pattern would remove some redundant lines of code (the `export const` declarations) and redundant imports.

On the other hand, I could see this pattern feeling easier to document.

But anyway, I just wanted to share some thoughts about it here (no change requested or suggested). Is this something I should open an issue to discuss, do you think you (and others) would be open to changing this pattern across the app?

Reply:
In the first iteration, I also had an array of constants for the filters state (`default` or `shown_on_desktop`, `visible` and `dismissed`). Then, I had to rename one of them, and saw that refactoring without constants is much more difficult: it is much easier to miss one of the usages because you use "search and replace" instead of the IDE tooling to `refactor`/`rename`. This is why I converted the strings into constants: for ease of refactoring both in the main code and in tests.

I don't have a very strong preference for this pattern, so I would love to hear everyone's opinion.

Reply:
We might not really need the constants here because these specific constants are not used in many places.